core/src/main/java/org/mapstruct/Javadoc.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.CLASS)
```

Comment:
Why `RetentionPolicy.CLASS`?

Reply:
Sorry @filiphr , I don't fully understand you here. Please, can you elaborate?

Reply:
The java compiler uses this `RetentionPolicy` to decide what to do with the annotation in the byte code. There are 3 retention policies:

* `SOURCE` - Annotations will be discarded by the compiler
* `CLASS` - Annotations will be recorded in the class file
* `RUNTIME` - Annotations will be recorded in the class file and retained by the VM at runtime

I think that for this particular annotation we can use the `SOURCE` retention policy, since we only need it when generating the code for a particular mapper.

Reply:
Obvious, sorry @filiphr.